tests/framework/extensions/clusters/clusters.go

```diff
@@ -1007,7 +1006,24 @@ func WaitClusterToBeUpgraded(client *rancher.Client, clusterID string) (err erro
 		return
 	}
 
+	return
+}
+
+// WaitClusterUntilUpgrade is a helper function that takes a rancher client, and the cluster id as parameters.
+// Waits until cluster is ready.
+// Cluster error states that declare control plane is inaccessible and cluster object modified are ignored.
+// Same cluster summary information logging is ignored.
+func WaitClusterUntilUpgrade(client *rancher.Client, clusterID string) (err error) {
```

Comment:
nit: Shouldn't this function be named something like "WaitUntilClusterReady" or something? Since it's waiting for a Ready status

Reply:
Although I understand your point, I feel like this is gonna cause confusion between `IsProvisioningClusterReady` and this. This is more customized to after-cluster upgrades/updates because of this `isClusterInaccesible` function where we make some error exceptions that happen in upgrade calls.